Empire/Empire.cpp

```diff
@@ -1480,8 +1519,10 @@ std::string Empire::Dump() const {
     return retval;
 }
 
-void Empire::SetCapitalID(int id)
-{ m_capital_id = id; }
+void Empire::SetCapitalID(int id) {
+    m_capital_id = id;
+    m_source_id = id;
```

Comment:
I've added a check that the source id is valid.  Source ids don't always need to exist in the code, it seems.